Basic theming and app icons

Cipher73 opened the pull request:
- Implemented SettingsScreen for theme management
-  Integrated shared preferences for persistent theme settings.
- Created basic BookSelectionScreen for selecting books with a simple UI
- Developed Basic HomeScreen with navigation to BookSelectionScreen and settings
- Removed unused second_screen.dart file

jeremyscodes committed:
update kotlin version

jeremyscodes committed:
Merge branch 'main' into app-icons

jeremyscodes committed:
staging icon

jeremyscodes committed:
production icons

Cipher73 committed:
feat: Implement Theme Management

- Created BookSelectionScreen for selecting books with a simple UI.
- Developed Basic HomeScreen with navigation to BookSelectionScreen and settings.
- Implemented SettingsScreen for theme management
- Integrated shared preferences for persistent theme settings.
- Removed unused second_screen.dart file.

Cipher73 committed:
added a simple test to find text on the settings screen

Cipher73 committed:
fixed linting problems

Cipher73 committed:
Merge branch 'app-icons' into basic_theming_AN

Cipher73 committed:
chore: Update dependencies in pubspec.lock

jeremyscodes committed:
Merge branch 'develop' into basic_theming_AN

jeremyscodes commented on lib/screens/home/home_screen.dart:
```diff
@@ -0,0 +1,92 @@
+        backgroundColor: context.appBarColor,
+        title: Text(
+          widget.title,
+          style: TextStyle(
```
Can we use a predefined theme text here instead of defining a TextStyle?

jeremyscodes commented on lib/screens/home/home_screen.dart:
```diff
@@ -0,0 +1,92 @@
+                      title: widget.title,
+                    ),
+                  ),
+                );
```
Can we update this to use the GoRouter pattern? We define the paths in main.dart

jeremyscodes commented on lib/screens/settings/settings_screen.dart:
```diff
@@ -0,0 +1,125 @@
+                            children: [
+                              RadioListTile<ThemeMode>(
+                                title: Text(
+                                  'Light',
```
Something I like to do in Flutter is extract these values (title and subtitle, in this case) and have them as an object outside of this scaffold and then read that object in a map.

Cipher73 replied:
Noted, definitely better for adding any other option because i would then put it on the list , Thank you 💮

jeremyscodes commented on lib/theme/app_fonts.dart:
```diff
@@ -0,0 +1,69 @@
+  AppFonts._();
+
+  /// Using Open Sans - a clean, readable sans-serif font perfect for books
+  // static const String _fontFamily = 'Open Sans';
```
Is this meant to be commented out?

Cipher73 replied:
yes , i was using that at first with a different font that i was trouble with , then decided to go with using Open Sans via the GoogleFonts package

will clean that and any other code i ended up not using

jeremyscodes reviewed (commented):
Amazing work! I've left a few comments. Let me know if you have any questions or disagree with anything.

Cipher73 committed:
resolve pr comments

- Cleaned up unused font family code in AppFonts.
- Introduced theme options in SettingsScreen for better user customization.
- Updated HomeScreen to navigate to BookSelectionScreen using GoRouter.
- Use predefined theme text instead of defining a TextStyle in home screen
- Refactored icon size handling with AppIcons utility.

Cipher73 committed:
Merge branch 'basic_theming_AN' of https://github.com/Family-Altar/family-altar-mobile into basic_theming_AN

jeremyscodes reviewed (approved):
Epic PR!
